Proposal review issue fix spree

lod opened the pull request:
Did an audit of all the proposal and review based issues.
Have addressed (fixed or commented on) all the review based issues.
Have also fixed all the proposal low hanging issues.

ben-denham committed:
Fixed missing tag in db_content edit form.

Commit:
Merge of Terms and Conditions fix into zookeepr master

steven-ellis committed:
Document the process for building the conference schedule

steven-ellis committed:
Add in info for daily schedule

steven-ellis committed:
Add readme for PaymentExpress process along with test credentials, refs RM 122

steven-ellis committed:
Document DPS field TxnId needs to be unique issues

steven-ellis committed:
Document the Rego Process

steven-ellis committed:
Automate process for uploading Video Links

steven-ellis committed:
Document automated process for uploading Video Links

KathyReid committed:
Initial revision, did first version of Tools section

Commit:
Ignore generated gid and idutils tags

KathyReid committed:
Moved .html files to .md files - they are MarkDown, not HTML files....

KathyReid committed:
removed stylesheets, not used with MarkDown

iseppi committed:
Merge pull request #318 from KathyReid/docs2016

Docs2016

iseppi committed:
* Add missing version
 * Fix sqlalchemy where clause

Commit:
Merge pull request #321 from iseppi/db-fixes

Various DB fixes

KathyReid committed:
Added images for comparing branches when issuing a pull request

jhesketh committed:
Merge pull request #317 from steven-ellis/master

 Add video_release information to JSON

jhesketh committed:
Merge pull request #320 from fmarier/ignore-tags

Ignore generated gid and idutils tags

KathyReid committed:
Finished section 01-Development

jhesketh committed:
Fix pydns and development.ini

jhesketh committed:
Merge pull request #323 from jhesketh/working

Fix pydns and development.ini

jhesketh committed:
Allow users to be logged in unactivated

Make the sign up process easier by not requiring users to be activated
to be logged in. Require an activated account for paper submission and
conference registration. Warn on granting privs to in-active account.

For the moment redirect to an info placeholder. Soon to be replaced with
fancy callback stuff.

fmarier committed:
Enlarge the hashed fields for SHA-256

fmarier committed:
Switch from MD5 to PBKDF2 for password hashes

fmarier committed:
Update the nasty hack to add support for the new field sizes

jhesketh committed:
Merge pull request #324 from fmarier/stronger-password-hashing

Stronger password hashing

iseppi committed:
Merge pull request #322 from KathyReid/docs2016

Docs2016

iseppi committed:
* Add development .wsgi file

jhesketh committed:
Capture email errors and display warning

jhesketh committed:
Go straight to registration on login

Rather than the rego status page, go straight to the rego form
if logged in and not registered.

Go toe CFP page if open (and rego closed)

Log in straight after registration

iseppi committed:
* Use os.path.normpath and os.path.join

jhesketh committed:
Merge pull request #325 from iseppi/add-wsgi

 * Add development .wsgi file

jhesketh committed:
Only grant role permissions to activated uesrs

jhesketh committed:
Warn about not-activated accounts on each page

steven-ellis committed:
Typo fixup on Rego guide

KathyReid committed:
Modified after verifying the process, and making some minor changes to list dependencies and gotchas. Things that still need to be done are flagged with @TODO

jhesketh committed:
Fix up permissions/access across controllers

fmarier committed:
Add an example of how to auth against the ZK DB in PHP

jhesketh committed:
Merge pull request #330 from fmarier/php-auth-example

Add an example of how to auth against the ZK DB in PHP

KathyReid committed:
fixed indentations

KathyReid committed:
added registration details from David Tulloh

KathyReid committed:
added registration details from David Tulloh

dtbell91 committed:
created production sample config files

noisymime committed:
Merge pull request #333 from dtbell91/sample-configs

created production sample config files

KathyReid committed:
added index to administration doco

fmarier committed:
Add a new DB field for the per-user salt

fmarier committed:
Add a random per-user salt value to the password hashes

fmarier committed:
Update the PHP example to use the per-user salts

fmarier committed:
Update the nasty hack to add support for the new password_salt field

jhesketh committed:
Redirect back to where the user wanted to go

As soon as the account is activated, continue on the next step.

lod committed:
Shift lca_info data into database

Created a database table config to act as a key store
Shift lca_info and lca_rego data into database table config
Required SQL is in zkpylons/config/lca_info.sql

Created a model, Config (config.py), to retrieve config values
Updated all uses of lca_info and lca_rego to use Config
Templates access Config via c.config

Reverted 8dc83a and rehardcoded the theme into zkpylons_config.py
We can't access Config at this point as the model has not been loaded

The submenus and menus variables are not a good fit for the keystore.
They have been moved to zkpylons/config/menu.py

Web based access to update the config values is possible but not
yet implemented.

jhesketh committed:
Merge pull request #335 from fmarier/user_salt

User salt

jhesketh committed:
Move enabled theme setting into lca_info

Commit:
Initial commit of LCA2016 theme

noisymime committed:
Merge pull request #337 from noisymime/master

Initial commit of LCA2016 theme

fmarier committed:
Replace PBKDF2 with SHA-256 (fix #336)

Ubuntu 14.04 doesn't have Python 2.7.8 (where pbkdf2 got added) :(

Commit:
Merge pull request #338 from fmarier/pbkdf2-to-sha256

Replace PBKDF2 with SHA-256 (fix #336)

Commit:
Sidebar fixes on LCA2016 theme

Commit:
Merge pull request #327 from steven-ellis/documentation_updates

Documentation updates from LCA 2015 Team

Commit:
Merge pull request #328 from steven-ellis/19_db_content_form_fix

Fixed missing tag in db_content edit form.

Commit:
Merge pull request #331 from steven-ellis/16_automate_video_link_uploads

16 automate video link uploads

Commit:
Merge pull request #334 from KathyReid/docs-2016

Docs 2016

fmarier committed:
More changes from Kathy

This is a rebase of the changes made in 29c8bc8ef94c40a377055d52b1688c4ee4c8c00b.

fmarier committed:
Remove INSTALL to INSTALL.md

Commit:
Merge pull request #329 from steven-ellis/25_terms_and_conditions_fix

Merge of Terms and Conditions fix into zookeepr master

Commit:
Merge pull request #326 from jhesketh/working

Working

Commit:
Bootstrap based form validation (For new person only at this stage)

Commit:
Merge remote-tracking branch 'zookeepr/master'

noisymime committed:
Merge pull request #340 from noisymime/master

Further integration with bootstrap for lca2016

Commit:
Further work on converting templates to boostrap

dtbell91 committed:
Merge pull request #343 from noisymime/master

Further work on converting templates to boostrap

lod committed:
Changes to Config.get() suggested by code review

Calling structure
Config.get(key) & Config.get(cat, key)
changed to
Config.get(key) & Config.get(key, category=cat)

Config.get() log level changed from info to warning.

lod committed:
Fixed bug config related bugs

Fixed bug in helpers accessing Config
c.config.get() not accessible by the helper, switch to Config.get()

Bug fixes on config string comparison in registration controller
Config.get() returns a unicode string so 'is' comparison fails
'is not' operators should be !=

No other similar instances were found in the code base

lod committed:
Created admin config display page

Created admin/change_config to display edit page
Created admin/config to provide JSON API for config variables

admin/change_config is an angular based page
This uses a custom base template, angular.mako.
angular.mako is based off base.mako and provides the standard styling
angular.mako also uses a custom routing solution based in part on
RequireJS. Existing routing solutions didn't meet the needs of dynamic
loading while not switching to single page view. This does mean that
all angular based code must be comply with RequireJS's AMD.
The framework should allow for a future evolvement to a single page
structure and/or automated javascript consolidation and minification.

Added description field to config keys table
This optional field is displayed when user edits the value.

Converted invoice/new to use angular.mako
invoice/new was created several years ago without styling
It now feels like a standard Zookeepr page
The typeahead functionality was disabled however. This was provided
by Bootstrap 2.0 but was dropped from 3.0

Further work required:
* Reenable typeahead functionality to invoice/new
* Convert JSON API to REST compliant structure

noisymime committed:
Remove mandatory status of mobile phone field

noisymime committed:
Further work to LCA2016 theme and CFP requirements

noisymime committed:
Fix for incorrect relative path on boostrap submenus

noisymime committed:
Merge pull request #373 from noisymime/master

LCA2016 theming and minor fix for default data

lod committed:
Zookeepr example API Documentation

readme.md is an overall of the API, its conventions and documentation.

configs.md is an example API implementation.

dtbell91 committed:
Assigned table class to tables on admin pages to improve readability. Also added sorttable.js which allows in browser sorting of tables.

dtbell91 committed:
Generate top banner news only from content before the break in news items

dtbell91 committed:
Moved changes to lca2016 theme folder until all templates are converted to bootstrap

lod committed:
Fixing model unit testing

In the upgrade of Pylons and partial shift to Pyramid the testing
framework got broken.

I tried hard to fix the old framework but the complexity of it defeated
me. It also seemed excessive to the actual project's needs.

So this is a rework of all the old model tests onto a new framework.

The new tests are part of the zk project as this is where the models
reside.

Tests are executed by pytest, this seems to be the current tool of
choice. From zookeepr root, `py.test`

There is some pytest magic in zk/tests/model/conftest.py
This automatic include is noted in each test file

factory_boy is used to generate dummy data fixtures in fixtures.py
The library is a bit complex but generally a new fixture can be created
by copying an existing one.
Using this library greatly reduces the complexity of each test.

Tests are performed against a real Postgresql database, as this best
matches the deployed system. However switching the SQLite is a one line
change. Each test occurs within a transaction which is rolled back on
completion. This means no data is actually ever commited and tests
should be able to run in parallel.

The old model table tests were all deleted.
These defined the table in the test file, then used common code to
check that the SQLAlchemy definition matched the test definition.
In essence it was testing SQLAlchemy, so wasn't needed.

The old domain tests included an automatic and custom test portion.
The automatic test inserted some sample data into the table and
deleted in, verifying each step. Again this is just testing SQLAlchemy
so I got rid of the automatic portion.
The domain test files which had no custom tests were deleted.

dtbell91 committed:
Merge pull request #380 from dtbell91/master

Merged changes to readability of administration tables in lca2016 theme

dtbell91 committed:
Added new db_content type, Banner

dtbell91 committed:
Replaced news with banner content in lca2016 theme. Other modifications to lca2016 css to better suit small screens.

dtbell91 committed:
Merge branch 'master' of github.com:dtbell91/zookeepr

dtbell91 committed:
Various changes to get lca2016 ready for open of cfp

dtbell91 committed:
Merge pull request #385 from dtbell91/master

Various changes for lca2016

dtbell91 committed:
Centered footer on lca2016 theme

dtbell91 committed:
Merge pull request #386 from dtbell91/master

Centered footer on lca2016 theme

dtbell91 committed:
Made Australia default country on rego page, changed wording to be consistently "terms and conditions"

dtbell91 committed:
Merge pull request #387 from dtbell91/master

Made Australia default country on rego page, changed wording to be co…

dtbell91 committed:
fixed sitemap link on lca2016 theme

dtbell91 committed:
Merge pull request #388 from dtbell91/master

fixed sitemap link on lca2016 theme

noisymime committed:
Merge remote-tracking branch 'zookeepr/lca2016'

dtbell91 committed:
add analytics tracking

noisymime committed:
Bootstrapped miniconf proposal

noisymime committed:
Merge branch 'zookeepr/lca2016'

Conflicts:
	zkpylons/themes/lca2016/templates/base.mako

noisymime committed:
Neatened up profile view page

dtbell91 committed:
Merge pull request #389 from noisymime/master

Bootstrapped miniconf submission form

noisymime committed:
Fix toolbox css weirdness for paper reviewers

dtbell91 committed:
Merge pull request #390 from noisymime/master

Fix toolbox css weirdness for paper reviewers

noisymime committed:
Allow profile editing without mobile number

dtbell91 committed:
Merge pull request #391 from noisymime/master

Allow profile editing without mobile number

dtbell91 committed:
Merge github.com:zookeepr/zookeepr

dtbell91 committed:
Merge github.com:noisymime/zookeepr

dtbell91 committed:
Fixed missing email address on person update bug

noisymime committed:
Merge pull request #392 from dtbell91/master

Fixed missing email address on person update bug

noisymime committed:
Merge pull request #1 from zookeepr/lca2016

Merge with lca2016

noisymime committed:
Prohibit changing email unless an admin

dtbell91 committed:
Shrunk images for lca2016 theme to improve page loading times

dtbell91 committed:
Merge pull request #394 from dtbell91/master

Shrunk images for lca2016 theme to improve page loading times

dtbell91 committed:
Remove old large .png images after converting them to .jpg

dtbell91 committed:
Made List of DB Pages table sortable

dtbell91 committed:
Merge pull request #395 from dtbell91/master

Removed old images, made list of db pages table sortable

dtbell91 committed:
added ie10-viewport-bug-workaround.js to lca2016

dtbell91 committed:
Merge pull request #396 from dtbell91/master

added ie10-viewport-bug-workaround.js to lca2016

dtbell91 committed:
fixed link to ie10-viewport-bug-workaround.js in lca2016

dtbell91 committed:
Merge pull request #397 from dtbell91/master

fixed link to ie10-viewport-bug-workaround.js in lca2016

chrisjrn committed:
Changes handling of t-shirt product names

Behaviour is now:

- The size is now the last token of the name (handles e.g. ```Men's M```) unless the second last word is "Size", in which case it's the last two (handles e.g. ```Women's Size 22```)
- The words leading up to the size are the style type (allows for categories such as ```Men's``` and ```Women's```; or ```Men's/Straight Cut```, ```Women's Fitted``` and ```Women's Semi-Fitted```
- Renames the ```gender``` variable to ```style```

chrisjrn committed:
oops

noisymime committed:
Merge branch 'lca2016' of https://github.com/zookeepr/zookeepr

noisymime committed:
Fix for inability to edit proposals

noisymime committed:
Fix for errors when editing normal proposal

noisymime committed:
Revert "Prohibit changing email unless an admin"

This reverts commit ae153abe7d088f9f210c59856345a0b4d2001cdf.

noisymime committed:
Paper review templates updated to bootstrap

noisymime committed:
Merge branch 'master' of https://github.com/noisymime/zookeepr

dtbell91 committed:
Merge pull request #393 from noisymime/master

Multiple fixes for editing miniconf and regular proposals

dtbell91 committed:
Added "required" labels to proposal forms

noisymime committed:
Merge pull request #402 from dtbell91/master

Added "required" labels to proposal forms

jhesketh committed:
Fix mistake in install

Database is expected to be zookeepr

iseppi committed:
* Add initialise Data script

jhesketh committed:
Merge pull request #404 from jhesketh/install_fix

Fix mistake in install

jhesketh committed:
Merge pull request #405 from iseppi/alembic

 * Add initialise Data script

neillc committed:
Rename alembic revisions to make ordering explicit

neillc committed:
Merge branch 'alembic/rename_revisions'

neillc committed:
Populate database with initial values

neillc committed:
Merge branch 'alembic/import_data'

jhesketh committed:
Merge pull request #406 from neillc/master

Clean up alembic and load initial data

jhesketh committed:
No longer need to initialise data

Data is now loaded from alembic.

jhesketh committed:
Remove initialise data script

No longer needed as all data should be set up and modified
via alembic.

jhesketh committed:
Fix depreciated argument in alembic

UserWarning: Argument 'name' is now named 'new_column_name' for function 'alter_column'

jhesketh committed:
Update install instructions for external deps

jhesketh committed:
Tidy up install file

jhesketh committed:
Merge pull request #407 from jhesketh/tidy_install

Tidy install

jhesketh committed:
Merge pull request #383 from lod/testing_fixup

Fixing model unit testing

jhesketh committed:
Set up tox and travis for new tests

jhesketh committed:
Copy lca_info for travis

jhesketh committed:
Merge pull request #408 from jhesketh/tox

Set up tox and travis for new tests

jhesketh committed:
Mark down the README

jhesketh committed:
Merge pull request #409 from jhesketh/markdown

Mark down the README

jhesketh committed:
Fix install link

jhesketh committed:
Merge pull request #410 from jhesketh/markdown

Fix install link

jhesketh committed:
Fix readme and config link

jhesketh committed:
Merge pull request #411 from jhesketh/markdown

Fix readme and config link

lod committed:
Clean out references to lca_info.py

Remnants (menu entries) are in zkpylons/config/menus.py
These don't have to be edited to get a working install.

lod committed:
Update lca2016 templates to use DB lca_info

Some were changed between the initial and updated commit, re-fix them.

lod committed:
Convert lca_info.sql into an alembic script

This attempts to migrate data from lca_info.py if it exists.
On a downgrade the database contents are exported to rebuild lca_info.py
Both directions should be reviewed by human, just in case.
Pretty messages are thrown at the user in both instances.

If it doesn't exist we use a default data set (built in).
We also reset the default password because we change the seeds.

lod committed:
Fix tests so that they work

Several problems needed to be addressed:
 * The config table had to be populated with the password_salt
   for anything touching the Person model to succeed.
   Everything touches the Person model.
   I introduced a bonus dependency here, fake-factory, because why not.
 * The Session variable was being copied before it was initialised.
   This object was then reused, causing errors.
 * zk.model.meta.Session and zkpylons.model.meta.Session are different
   and Python seems to randomly choose which one it uses at any given
   time. Both must be initialised for test code to work.

lod committed:
Fixed root cause of zk/zkpylons switch issue

Previous commit plastered over a problem where Python was switching
from zk/model/ to zkpylons/model/, which caused a switch between
Session globals, which caused really odd failures.

Root cause was fairly easy to understand when I found it. I had
specified the absolute module, zkpylons.model.Config rather than the
relative Config. Almost certainly due to copy/pasting the line from
elsewhere in the code base.

This minor patch fixes model/person.py to use a relative module.
Grep strongly indicates that no other files suffer from the same
problem.

I am leaving the prior workaround in place however as I suspect this may
crop up again.

lod committed:
Improve JSON editing support on admin config page

Old page provided complex JSON objects as text in a single line text
input. It had verification for when you stuffed up, but essentially you
had to copy-paste out, reformat, edit, reformat and paste back in.
v1.0 funtional

New version magically opens large strings such as JSON stuff out to a
multiline box. JSON is indented before you get to it, life is better.

Life has additional complications because contenteditable is a hairy
mess. Looks sweet though.

lod committed:
Deleted unused or invalid tests

Openday isn't a ZK concept any more
Scaffold test - not used
Not using the unittest framework
Remove test_template - doesn't test anything
test_proposal_type is covered by test_proposal
Only validator that was tested is gone
zkpylons.lib.crud is gone

lod committed:
Changed missing role behaviour

If a role is requested but not in the database the code threw an
exception. I have changed this to throw a NotAuthorizedError.

This makes the failure a graceful rather than terminal one.
The change is being done to make testing easier, it should also help
development allowing code to be written before the database change is
made.

There is no impact on a production site, the code should not run.

lod committed:
Rewrite tests to allow them to work again

Old tests used a complex framework, which has been entirely discarded.
A new hopefully less complex framework is used instead.

Tests also transition from nose, I think, to pytest runner.

Some tests also require significant work to reflect the reality of the
system. Many are unchanged since 2006.

Some tests fail, these issues raised in the github tracker.
The issue number is marked in the test, the tests have been flagged as
an expected fail for this merge. This flag should be removed when the
issue is addressed.

lod committed:
Minor fix in HTML

We had an excess </p> making the HTML invalid.
Not a concern for users but made automated testing have conniptions.

lod committed:
Fix #418, assuming that first/last name is set

Mostly switch to fullname to avoid issue

lod committed:
Fix #416, proposal validation requiring phone

Also dropped some related duplicate code

lod committed:
Fixes #417, thankyou breaking on no email set

lod committed:
Reduced number of HTML parsing errors

Automated testing of registration form was failing.
It wasn't picking up all the elements.

I suspected that the broken HTML might be putting the parser off.

Reduced the HTML errors from 185 to 145.
Including hitting most of the structural and all of the form issues.

lod committed:
Fix SQLAlchemy detacted error

This has been plaguing the functional testing.
If the page does an internal redirect the Session is closed...
I assume under normal conditions it is opened as part of the redirect.
Under testing it is not, and odd errors ensue.

lod committed:
Fixed test

Unsure why it broke... :(

lod committed:
Added database wipe to DB tests

The tests don't commit, so it wasn't needed before.
But with the automated testing running the alembic setup the db gets
populated will all sorts of standard stuff that gets in the way.

lod committed:
Fix #419 - update test to match desired behaviour

Also renamed the test to reflect the new function.

lod committed:
Created a version of lca_info for testing

Mostly the default one but all rego and cfp are open.

iseppi committed:
Merge branch lca2016
 * Merge lca2016 theme
 * An alembic revision has replaced the changes to zk/model/db_content.py (If there is already a row in the db_content_type table with a name of Banner you will need to stamp to revision 3b5c7d3c8366
 * Changes to zkpylons/controllers/person.py and zkpylons/controllers/proposal.py have been excluded from this merge

iseppi committed:
* Remove setup() from all models

iseppi committed:
Merge pull request #423 from iseppi/remove_legacy_initial_data

Remove code used in old initial data loads

iseppi committed:
Merge pull request #422 from iseppi/merge-lca2016

Merge lca2016

lod committed:
Update lca2016 theme to use new config structure

lod committed:
Account for upstream changes

Upstream has a new revision, bump the counter to keep things neat.

jhesketh committed:
Update install instructions

Remove copying lca_info file

jhesketh committed:
Remove sitemap controller

Do people still use sitemaps? Since the menu will be in the theme, the
site map can be managed as a db-content page if people still want it.

Eventually if we implement a menu system we can add back in a dynamic map.

jhesketh committed:
Update alembic upgrade instructions

jhesketh committed:
Move menu into templates

Set/override the menu in your theme.

lod committed:
Partial implemented - stashing for later

There are changes to the menu files being introduced to master, this
test should be readdressed once those are settled.

jhesketh committed:
Merge pull request #424 from jhesketh/lod-config_gui

Lod config gui

lod committed:
Fix test execution with config table

Mostly involved create config table entries where required.

lod committed:
Fix #426

Recently introduced bug causing submenus not to be generated.

lod committed:
Fix #426

Recently introduced bug causing submenus not to be generated.

lod committed:
Finish test_menu.py

Fixes previous embarrassing placeholder commit

lod committed:
Added tests enabled by Config being in db

Also created CompletePersonFactory to simplify some tests

jhesketh committed:
Merge pull request #427 from lod/fix_426

Fix #426

jhesketh committed:
Merge pull request #377 from lod/api_rfc

Zookeepr example API Documentation

lod committed:
Removing last traces of lca_info

lca_info.py.functional test was required to create a testing version of
lca_info.py, back when lca_info.py was required.

This is no longer the case so the file can go bye bye.

lod committed:
Cleaned up db initialisation

Pull the db initialisation string from the test ini file, this matches
what the application initialisation does so we have a single source.

Also cleaned up the population of the initial config table values for
the functional tests.

lod committed:
Add missing dependency - reportlab

Fixes #428

lod committed:
URL permission test/audit

This extensive (but not comprehensive) test hits a large number of urls
with each of the different roles that we use.

The response is then compared to a known "good" table to ensure that the
permissions are correct.

In practice the "good" table is our current permission set. This test is
designed to provide a backstop and prevent any inadvertant loosening or
regression in permissions.

It also provides a useful overview of the state of the system.

lod committed:
Fixed issue with inadvertent logging out

Was hitting the logout page mid-test, invalidating half the results.
This also fixes the consistency issues I was having.

jhesketh committed:
Move theme into a config variable

jhesketh committed:
Merge pull request #420 from lod/func_testing

Fix/reenable zkpylons functional testing

lod committed:
Fix #28 - make reviewing a single step

Old process:
	1. Open list of proposals needing review
	2. Open a specific proposal
	3. Click review link
	4. Perform review

New process:
	1. Open list of proposals needing review
	2. Open a specific proposal
	3. Perform review

This was achieved by merging the review page into the view page.
It is actually a fairly minor merge as the review page template already
included the view page template in its entirety.

lod committed:
Picky whitespace alignment

lod committed:
Template cleanup and simplify

new.mako and edit.mako were close enough to identical that I new.mako
could be deleted and edit used in its place.

form.mako got incorporated into edit as the abstraction was no longer
required.

lod committed:
Test #31 fixed - Editing reviews retains values

Bug report, from 2009, that editing a review retained text fields but
not combo or radio boxes.
Possibly fixed a few months later in 2009

This updates test_edit_review to explicitly test for it to prevent
regressions.

lod committed:
Fix template category titless

When listing proposals an s is added to the end of the category to
pluralise it. This looks rather silly if the category title already ends
in an s.

So now adding the s is only done if there isn't already one there.

This patches over the issue somewhat, proper pluralisation is hard.
Rails trys rather hard to do it and there are always bugs, the english
language is just too fucked up to do it properly.

We kind of get away with it as the categories don't change often and the
general public don't see it.

lod committed:
Fix #171 - Add page based navigation menu

Discovered that there was actually already a navigation menu generated
however it wasn't being shown in the default theme, or other themes.

This also enables context based navigation on a bunch of other pages
that have been generating it.

lod committed:
Clean up proposal review related templates

proposal/list_review.mako and proposal/summary.mako were a bit crusty
with complex mako and python code, db queries that weren't being used
and some scary ugly Javascript. review/summary.mako was less bad but I
hit it in the same sweep

Reworked the controllers, got rid of the setattr/getattr dynamic model
to make the data passing more explicit, pushed much of the python code
from the template back up into the controller and deleted code that was
collecting data that was never used.

Reworked the templates to take into account the controller changes,
fixed up the indenting, pushed some styling from element tags and inline
out to the css files, reworked Javascript to use a little bit of jQuery
and be far less scary and ominous.

Related work:
* Removed the stats_query and related stuff from the models, rather ugly
  sql alchemy magic and it was very task specific violating the
  encapsulation layers. Replaced it with proposal.average_score()
* Reworked helper line_break so that it can be used as a mako filter and
  not require the naked/unsafe output with filtering disabled.

Inadvertent improvements:
* /review/summary now shows all reviewers, including those who have not
  yet done any reviews
* /proposal/summary now shows all proposals, including those which have
  not yet been reviewed

lod committed:
Convert h.line_break to be markupsafe

This allows h.line_break to be used as a filter, as well as as a
function.

The main advantage of this is that it allows the standard html escaping
filter to be applied, rather than requiring the naked flag (|n) to be
used. This improves #432.

I changed all the simple mako line_break invokations to use the filtered
style, I have not verified all these by hand.

In practice this makes things easier and fixes a number of issues:

* Some code manually escaped the html before using line_break and the
  naked filter. This was made considerably easier to read.

* Some code used line_break but did not enabled the naked filter. This
  would have caused the added <br> to be escaped and show up literally
  in the output... this is now fixed.

* Some code used line_break with the naked filter but did not manually
  html escape the string first, causing a potential injection attack
  site... this is now fixed.

lod committed:
Fix up h.url_to_link

Modified h.url_to_link to work with markupsafe, this allows it to be
used in a filter and improves the safety of the code base from injection
attacks. This improves #432.

Also improved/simplified the regex to fix a few of issues:
* [url=https://...] would not be parsed, only http:// was supported
* Multiple url tags would not be parsed, first match was greedy and
  attached to the last.

+ unit tests

lod committed:
Fix bug reported by dtbell - reviewer cannot load

Angular template did not provide define structures for the reviewer or
funding reviewer which is used by the toolbox template.

lod committed:
Make h.yesno() not require base output

Made h.yesno() flag its ouput as safe so that the naked filter disable
flag is not required to use it. This improves #432 by 60 entries.

lod committed:
Fix #303 - stalk link issues

The email stalk links weren't showing
Some of them were also broken - not searching properly
Also fixed indenting and some html errors

lod committed:
Use javascript to hide biasing info from reviewers

Fixes #77 - Hide other reviewer scores
Note that this bug references two pages, previous work has
combined them into one so only one area needs to be fixed.

Fixes #168 - Hide other reviewer scores
Slightly different detail, not implemented as requested but
I think it meets the goals.

Fixes #169 - Hide accommodation and travel assistance requests
Like #168 this is done in a slightly different manner to the
one proposed, but I feel it meets the requirements.

lod committed:
Investigated next proposal and tweaked it

Issue #8 requests that the next random proposal be of the same type
as the current proposal. I confirmed that this is already the case.

Wrote a test for zk.model.proposal.find_next_proposal() locking in the
current behaviour.

Improved the SQL query in find_next_proposal(), removed some elements
which were used in the past but are no longer required - roughly doubles
the speed of the query.

Added another filter to the SQL query so that next proposal can't be
authored by the reviewer. Costs roughly half of the previous gains.
This makes the next proposal consistent with the proposal listing.

Changed the calling syntax for find_next_proposal() to make it
consistent with the other model functions. Old syntax was:
prop.id = Proposal.find_next_proposal(prop.id, prop.type.id, person.id)
New syntax uses objects in and out rather than ids:
prop = Proposal.find_next_proposal(prop, person)

lod committed:
Fix #306 - review comment columns too small

The issue is a little more complex than reported.
The table listing the review comments was a variable width table that
took the full width of the active body area. Normally that meant that
the comments took roughly a third of the width, private comments another
third. However if one column had a long string in it, like a link, then
it stole column width from the others leading to a very narrow column as
depicted in the bug report.

However, even at its optimum the arrangement is poor. The area to enter
comments is substantial encouraging a paragraph or two. We want to
encourage this level of detail. Having a third of the active area given
to the comments is poor and rather hard to read.

This commit fixes all this by abusing the standard table model.
Each review gets three rows of the table. The first row has the
reviewer, score and (optionally) the stream. The second row has the
comments, the third the private comments.

The table is abused further by making the comments use up all the table
columns except the first. So typically the comments section takes over
five columns, 90% of the active area width.

In practice comments are now nice solid paragraphs and the world is a
better place.

jhesketh committed:
Merge pull request #433 from lod/change_config_reviewer_bug

Fix bug reported by dtbell - reviewer cannot load

lod committed:
Fix #76 - Detect duplicate submissions

This commit detects proposals with similar titles and abstract texts.
The test is performed when a reviewer loads the proposal, if any
potentially duplicate proposals are found a list is provided to them.

This involved putting a new function in the proposal model, adding the
functionality to the controller and template. Both model and functional
testing are provided.

The comparisons are performed using postgresql functions which are
provided with the program but typically packaged as extras. The alembic
version enables these function groups.

Debian based systems will need to install postgresql-contrib-9.4 or
similar. Other distributions may also require the contrib or extra
package.

A fallback is also provided for non-postgresql systems using identical
matches.

Performance, benchmarking and optimisation was performed against the
2016 submission set.

jhesketh committed:
Rename setup method to initilise

jhesketh committed:
Awful hack to fix tests

This will be updated/fixed when the new auth system is done.

It's hacky, but consistent with the rest of the tests. Rather than
mocking out the config, assume if a page gets a 500, it's passed the
auth and is equiv to a 200.

jhesketh committed:
Merge pull request #431 from jhesketh/theme_config

Move theme into a config variable

lod committed:
Fixed #33 - sort by type on /proposal/approve

Issue requested sorting by type on the basis that updates were typically
applied to one type at a time.

Initial fix was to just apply a sort.

Then transitioned to providing a heading and table per type, this is
clearer to use, more consistent with other pages, and allowed the easy
creation of a contents menu for faster navigation.

lod committed:
Fix #166 - Remove miniconf links after approval

The issue is that after miniconfs are decided they are still listed on
the to-be-reviewed page.

I updated the choose review page to only list proposals that had a
status of pending review. Also updated find_next_proposal to only choose
pending review status too.

To wrap things up any proposal type without an eligible pending proposal
is hidden.

To prevent this being confusing the review summary page also displays
the status as a score if it isn't pending review.

lod committed:
Stumbled across find_accepted_by_id

Decided to clean it up a bit and removed the function.

Then spent considerably more time writing a unit test.

lod committed:
Fix #78 - Improve experience title wording

lod committed:
Fix #435 - prepopulate mobile phone field

lod committed:
Fix #398 - optional private abstract

Private abstract was a mix of required and not.
Having it required doesn't really make sense, so now it is consistently
not.

lod committed:
Fix #304 - change advised abstract length

Issue suggests changing maximum length from 500 to 250.
2016 proposals show a mode=105, median=138, mean=157

I feel suggesting a length rather than a limit is better guidance.

jhesketh commented:
Wow, you're a rocket. Nice stuff.

I've skimmed this, but not enough to merge it. That said, it's a pretty harmless change given that the CFP and reviewing periods are over. If somebody else can skim/review that'd be good. Otherwise a manual confirmation that it works for somebody else and then we should merge (before bitrot).

jhesketh commented:
@iseppi - ping, care to take a look at this at some point?

lod committed:
Create lca2016 themed version of angular.mako

Fixes #422

dtbell91 committed:
Merge pull request #444 from lod/fix422

Create lca2016 themed version of angular.mako

jhesketh committed:
New deps from updated pillow

stickster committed:
Fix missing postal code error

lod committed:
Fix attachment permission bugs

jhesketh committed:
Merge pull request #479 from lod/fix_478

Fix attachment permission bugs

lod committed:
Added functional testing for CRUD like controllers

The tests are implemented using a CrudHelper class which the tests
inherit from.

CrudHelper implements the test suite:
 * test_permissions
 * test_new
 * test_index
 * test_view
 * test_edit
 * test_delete

Each function takes as its required parameters: self, app, db_session
Each function also takes a large variety of optional parameters,
these allow the function to be tweaked and customised depending on the
expected output. Common customisations are things like a different form
of page title or a custom target object to test for.

There are three ways these functions can be used by each test file:
 1. No code written, the generic test in inherited and used
 2. Customisation code written, the generic test is called from
    within the test file function of the same name providing customised
    options and typically extra code before or after.
 3. CrudHelper code overwritten, the test file has a function of the
    same name but does not call the CrudHelper function. This allows
    full custom testing but is typically used to disable the test if the
    controller does not support the function.

Improved diagnostic of mako errors
----------------------------------

The best solution is to change the stack trace so you get a nice Mako
based trace rather than the 30 odd full list

This was the prior behaviour provided by pylons.error handle_mako_error

Unfortunately it doesn't work, I spent hours trying to work it out but
it seems that the stack trace is being rewritten again by one of the
third party layers.

So for now we do the stack rewrite - it might start working

We also print out the nice Mako stack trace so we have something to work
with in the mean time.

Testing improvements
--------------------

Make login function reentrant
This simplifies tests by not requiring a logout and state tracking

Bug fixes
---------

In addition to implementing the tests a stack of minor bug fixes have
also been applied as the bugs were uncovered.

This range from the very minor - changing the case of the template title
string to be consistent.

To the relatively major - existing functionality did not compile and was
obviously never tested.

Due to the large number of these changes they were not broken out in to
individually reported issues.

lod committed:
New alembic file populates missing lookup tables

Fills provisional data for:
    event_type
    fulfilment_status
    fulfilment_type
    fulfilment_type_status_map

iseppi committed:
* Order Fulfilment in FulfilmentGroup by the type_id

neillc committed:
Merge pull request #494 from iseppi/order-fulfilment-relation

 * Order Fulfilment in FulfilmentGroup by the type_id

neillc committed:
Merge pull request #446 from stickster/postcode-fix

Fix missing postal code error

lod committed:
Updated install process and instructions

Updated the install documentation.
Mostly just an edit pass.
Added a few sections and a step for development.ini

Also updated both tox.ini and setup.py to work with obselete
packages.

venusian was added to requirements, seems to be used by alembic.

lod committed:
Fix travis/tox issues on remote server

New versions of pip require the --trusted-host parameter
otherwise they die hard.
Old versions of pip die hard if --trusted-host is present.

Standard virtualenv such as Debian's uses an old version of pip
travis-ci.org which we use for acceptance testing uses a new version

The only solution I have managed to come up with is to have two files.
One providing a config for each system.
  tox.ini is the default and set for the old pip versions
  tox.pip18.ini is for new pip versions and is used by .travis.yml

lod committed:
Renamed tox.pip18.ini to tox.pip6.ini

Name reflects the minimum pip version required to switch files.

iseppi committed:
Merge pull request #499 from lod/fix_travis

Fix travis/tox issues on remote server

iseppi commented:
@lod could you rebase this and i'll try to get it merged.

iseppi committed:
Merge pull request #399 from chrisjrn/patch-1

Changes handling of t-shirt product names

iseppi committed:
Merge pull request #489 from lod/fix_486

fix #486 New alembic file populates missing lookup tables

lod commented:
@iseppi It isn't clear but I rebased it so it's clean now.

lod committed:
Try and fix travis CI - add pwgen

lod committed:
Deleting a stack of templates that aren't used

Most look like they were added at some point and then the functionality
was implemented differently but the template left behind.

Removing them decreases the maintenance load.

iseppi committed:
Merge pull request #506 from lod/unused_templates

Deleting a stack of templates that aren't used

iseppi committed:
Merge pull request #503 from lod/crud_tests

Added functional testing for CRUD like controllers

iseppi commented:
So, after merging your other two pull requests, this needs to be rebased again.

Also you will need to fix your alembic revisions to fit into the current tree.

lod commented:
Thanks for the merges.

Pull request has been rebased back on to master, had to force push because I was changing the history...
